radixdlt/src/integration/java/com/radixdlt/integration/distributed/deterministic/DeterministicNodes.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.integration.distributed.deterministic;
+
+import com.google.common.collect.ImmutableBiMap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.counters.SystemCounters;
+import com.radixdlt.integration.distributed.deterministic.network.ControlledMessage;
+import com.radixdlt.integration.distributed.deterministic.network.DeterministicNetwork.DeterministicSender;
+import com.radixdlt.utils.Pair;
+import io.reactivex.rxjava3.schedulers.Timed;
```

Comment:
Not convinced about this dependency.

Reply:
True, but I think good enough for now. Would rather hold off on making decision on how to handle timed messages.